__tests__/app.test.js

```diff
@@ -1,69 +1,221 @@
+                    });
+                });
+            });
+    });
+});
+
+describe("GET: /api", () => {
+    test("returns an object of APIs with their description", () => {
         return request(app)
-        .get('/api/topices')
-        .expect(404)
-        .then((response) => {
-            console.log(response.body)
-        expect(response.body.msg).toBe('Invalid URL');
-        });
-})
-})
+            .get("/api")
+            .expect(200)
+            .then((response) => {
+                expect(response.body.apiData).toEqual(apiEndpointsJSON);
+            });
+    });
+});
 
-describe('/api', () => {
-    test('returns an object of APIs with their description', () => {
+describe("GET: /api/articles/:article_id", () => {
+    test("GET: 200 an article object with all associated properties", () => {
         return request(app)
-        .get('/api')
-        .expect(200)
-        .then((response) => {
-                expect(response.body.apiData).toEqual(apiEndpointsJSON)
-            })
+            .get("/api/articles/1")
+            .expect(200)
+            .then((response) => {
+                const { article } = response.body;
+                expect(article.article_id).toBe(1);
+                expect(article).toMatchObject({
+                    created_at: expect.any(String),
+                });
+            });
+    });
+    test("GET:400 sends an appropriate status and error message when provided with a invalid id", () => {
+        return request(app)
+            .get("/api/articles/abc")
+            .expect(400)
+            .then((response) => {
+                expect(response.body.msg).toBe("Bad Request");
+            });
+    });
+    test("GET:404 sends an appropriate status and error message when provided with a non-existent but valid id", () => {
+        return request(app)
+            .get("/api/articles/999")
+            .expect(404)
+            .then((response) => {
+                expect(response.body.msg).toBe("Article does not exist");
+            });
+    });
+});
+
+describe("GET: /api/articles", () => {
+    test("GET: 200 returns an array of all objects within articles without body property, sorted in descending order by creation date", () => {
+        return request(app)
+            .get("/api/articles")
+            .expect(200)
+            .then((response) => {
+                const { articles } = response.body;
+                expect(articles.length).toBe(13);
+                articles.forEach((article) => {
+                    expect(article).toMatchObject({
+                        title: expect.any(String),
+                        topic: expect.any(String),
+                        author: expect.any(String),
+                        created_at: expect.any(String),
+                        votes: expect.any(Number),
+                        article_img_url: expect.any(String),
+                        comments_count: expect.any(String),
```

Comment:
It may also be worth checking the comments_count is correct for one of these articles :) just to show it works as expected